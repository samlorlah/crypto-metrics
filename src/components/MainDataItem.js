import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { IconContext } from 'react-icons';
import { BsArrowRightCircle } from 'react-icons/bs';
import styles from './MainData.module.css';

const MainDataItem = ({
  id,
  symbol,
  coinName,
  price,
}) => {
  const url = `/details/${id}`;
  return (
    <li>
      <div style={{ float: 'right', boxSizing: 'border-box' }}>
        <Link to={url}>
          <IconContext.Provider value={{ style: { fontSize: '20px', cursor: 'pointer', padding: '10px 10px 0 0' } }}>
            <BsArrowRightCircle />
          </IconContext.Provider>
        </Link>
      </div>
      <div className={styles.dataCategory}>
        <h5 className={styles.symbol}>{symbol}</h5>
        <div className={styles.nameAndPrice}>
          <h4>{coinName}</h4>
          <p>
            $
            { parseFloat(price).toLocaleString('en-US', { minimumFractionDigits: 2, maximumFractionDigits: 2 }) }
          </p>
        </div>
      </div>
    </li>
  );
};

MainDataItem.propTypes = {
  symbol: PropTypes.string.isRequired,
  coinName: PropTypes.string.isRequired,
  price: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
};

export default MainDataItem;
