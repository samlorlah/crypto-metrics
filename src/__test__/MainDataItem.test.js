import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import MainDataItem from '../components/MainData';

describe('Test Compenent that renders single list of coin', () => {
  it('Return a single card of of coin', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <MainDataItem />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
