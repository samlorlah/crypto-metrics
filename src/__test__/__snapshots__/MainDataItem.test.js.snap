// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Compenent that renders single list of coin Return a single card of of coin 1`] = `
<div
  className="dataContainer"
>
  <ul />
</div>
`;
